### PR TITLE
MULE-12866: HTTP Stream Response - remove switch

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: src/main/java/org/mule/extension/http/internal/request/HttpRequesterProvider.java [hunk 1]
```diff
@@ -106,6 +109,8 @@
 
   private TlsContextFactoryBuilder defaultTlsContextFactoryBuilder = TlsContextFactory.builder();
 
+  private boolean streamResponse = parseBoolean(getProperty(STREAM_RESPONSE_PROPERTY, "true"));
```
Comment (dfeist) [inline src/main/java/org/mule/extension/http/internal/request/HttpRequesterProvider.java]: Use `java.lang.Boolean#getBoolean`

Comment (afelisatti) [inline src/main/java/org/mule/extension/http/internal/request/HttpRequesterProvider.java]: Cleaner, but that default to false.

Comment (dfeist) [inline src/main/java/org/mule/extension/http/internal/request/HttpRequesterProvider.java]: Isn't it best to define the property such that non-existence of property is same as false anyway? i.e. `mule.http.disableResponseStreaming`

File: src/main/java/org/mule/extension/http/internal/HttpConnectorConstants.java [hunk 2]
```diff
@@ -20,5 +20,6 @@
   String CONNECTOR_OVERRIDES = "Connector Overrides";
   String TLS_CONFIGURATION = "TLS Configuration";
   String AUTHENTICATION = "Authentication";
+  String STREAM_RESPONSE_PROPERTY = "mule.http.streamResponse";
```
Comment (dfeist) [inline src/main/java/org/mule/extension/http/internal/HttpConnectorConstants.java]: Is `org.mule.runtime.core.api.config.MuleProperties#SYSTEM_PROPERTY_PREFIX` visable to use?

**Pull request merged**